File: problem/Min_Operations/minOperations.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function minOperations(nums) {
    const s = [];
    let res = 0;
    for (const a of nums) {
        while (s.length && s[s.length - 1] > a) {
            s.pop();
        }
        if (a === 0) continue;
        if (!s.length || s[s.length - 1] < a) {
            res++;
            s.push(a);
        }
    }
    return res;
};

File: problem/Min_Operations/minOperations.py
class Solution(object):
    def minOperations(self, nums):
        s = [0]
        res = 0
        for a in nums:
            while s and s[-1] > a:
                s.pop()
            if not s or s[-1] < a:
                res += 1
                s.append(a)
        return res